src/main/scala/edu/cmu/cs/obsidian/Main.scala

```diff
@@ -210,6 +212,54 @@ object Main {
         f.delete()
     }
 
+    def generateFabricCode(mainName: String, outputPath: Option[String], srcDir: Path): Unit = {
+        try {
+            //what we need to do now is move the .java class and the outterclass to a different folder
+            /* if an output path is specified, use it; otherwise, use working directory */
+            val outputPath_temp = outputPath match {
+                case Some(p) =>
+                    val path = Paths.get(p + mainName)
+                    /* create the dir if it doesn't exist */
+                    Files.createDirectories(path)
+                    path
+                case None =>
+                    val path = Paths.get(mainName)
+                    /* create the dir if it doesn't exist */
+                    Files.createDirectories(path)
+                    path
+            }
+
+            //copy the content of the fabric/java/ folder into a folder with the class name
+            val copyFabricFolderInvocation: String =
+                "cp -R fabric/java/ " + outputPath_temp
+            copyFabricFolderInvocation.!
+
+            //copy both the java and outerClass.java files to the Fabric folder
+            val javaSourceLocation = Paths.get(srcDir + "/edu/cmu/cs/obsidian/generated_code/" + mainName + ".java")
```

Comment:
Rather than concatenating, use the multi-argument version of Paths.get:

```
public static Path get(String first,
                       String... more)
```